server/channels/api4/outgoing_oauth_connection.go

```diff
@@ -328,3 +333,60 @@ func deleteOutgoingOAuthConnection(c *Context, w http.ResponseWriter, r *http.Re
 
 	ReturnStatusOK(w)
 }
+
+// validateOutgoingOAuthConnectionCredentials validates the credentials of an outgoing oauth connection by requesting a token
+// with the provided connection configuration. If the credentials are valid, the request will return a 200 status code and
+// if the credentials are invalid, the request will return a 400 status code.
+func validateOutgoingOAuthConnectionCredentials(c *Context, w http.ResponseWriter, r *http.Request) {
+	auditRec := c.MakeAuditRecord("validateOutgoingOAuthConnectionCredentials", audit.Fail)
+	defer c.LogAuditRec(auditRec)
+	c.LogAudit("attempt")
+
+	if !checkOutgoingOAuthConnectionWritePermissions(c) {
+		w.WriteHeader(c.Err.StatusCode)
```

Comment:
My understanding is that header codes are already taken care of by common [web handler](https://github.com/mattermost/mattermost/blob/a07097ed57cb54e59b801122ef21bf16426f7e97/server/channels/web/handlers.go#L393) in case an error is set through the context. Do we need the additional calls here and below?

Reply:
You are right. I just added a test to check this out, I don't know why I specifically added this.